File: kinotic-js/workspace/packages/management-api/src/api/model/grind/events/TaskFailedEvent.ts
import { JobRunEventType } from '@/api/model/grind/events/JobRunEventType'

/**
 * A task terminated with a failure.
 */
export class TaskFailedEvent {

    public readonly type: JobRunEventType.TASK_FAILED = JobRunEventType.TASK_FAILED

    /**
     * The task's position in the run's task tree.
     */
    public readonly taskPath: string = ''

    /**
     * The failure message.
     */
    public readonly error: string = ''

}

File: kinotic-grindv2/src/main/java/org/kinotic/grindv2/api/model/events/TaskFailedEvent.java
package org.kinotic.grindv2.api.model.events;

/**
 * A task terminated with a failure.
 *
 * @param taskPath the task's position in the run's task tree
 * @param error    the failure message
 */
public record TaskFailedEvent(String taskPath, String error) implements JobRunEvent {
}
